src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -156,6 +156,63 @@ public void ResolvesServiceMixedServiceAndOptionalStructConstructorArgumentsReli
             }
         }
 
+        public enum TheEnum
+        {
+            HelloWorld = -1,
+            NiceWorld = 0,
+            GoodByeWorld = 1,
```

Comment:
Ha!  I took the repro as a "rube goldberg" joke as how to make a more complicated HelloWorld that exposed the bug (kudos to the cleverness of @chrarnoldus).  I like that you embraced that in the test here.